### PR TITLE
Doxygen snippets were added for C/C++ files

### PR DESCRIPTION
**Pull request by PavloSolntsev**
Doxygen related snippets were added for C/C++ files to provide documentation for functions, enum, struct, and class. Also gpl3 snippet was introduced. If you like it I may add some other too.

### CODE REVIEW

**Comment by PavloSolntsev** [comment]
It looks like I added some extra spaces. Do I need to clean them and send another pull request? Thanks.

**Comment by PavloSolntsev** [comment]
Just pushed version with correction. Sorry, just learning how to work with people via git.

File: snippets/c.snippets [hunk 1]
```diff
@@ -228,7 +303,33 @@ snippet getopt
...
+# TODO section
+snippet todo
+	/*! TODO: ${1:Todo description here}
+	 *  \todo $1
+	 */
 ## Miscellaneous
 # This is kind of convenient
 snippet .
 	[${1}]
+# GPL v3
```
Comment (lpil) [inline snippets/c.snippets]: No need for this as we already have generic licence snippets

File: snippets/c.snippets [hunk 2]
```diff
@@ -125,11 +125,68 @@ snippet do
 ##
 ## Functions
 # function definition
-snippet fun
-	${1:void} ${2:function_name}(${3})
+#snippet fun
+#	${1:void} ${2:function_name}(${3})
+#	{
+#		${4}
+#	}
```
Comment (lpil) [inline snippets/c.snippets]: Looks like we lost a snippet here

File: snippets/c.snippets [hunk 3]
```diff
@@ -140,21 +197,39 @@ snippet td
...
 	typedef struct ${2:_$1 }{
-		${3:/* data */}
+		m_${3:Data} /*!< ${4:Description} */
 	} ${1:`vim_snippets#Filename('$1_t', 'name')`};
 
 snippet enum
+	/*! \enum $1
+	 * 
+	 *  ${2:Detailed description} 	
+	 */
```
Comment (lpil) [inline snippets/c.snippets]: Can we keep the original doxygenless ones and add new ones for the doced ones? i.e. `enumd`

File: snippets/c.snippets (hunk 2 above)
Comment (PavloSolntsev) [inline snippets/c.snippets]: Yes, think it make sense to leave it for backward-compatibility. Do you prefer name fun0, fun2 etc or we can add dfun0, dfun1 version for documentation purpose. Thus, we can leave funN versions without documentation and add dfunN version to include documentation. I personally, think that the present documentation block should slightly force people provide some explanation to the instance.

File: snippets/c.snippets (hunk 3 above)
Comment (PavloSolntsev) [inline snippets/c.snippets]: As I mentioned above, I would propose to add d-prefixed versions with documentation section.

File: snippets/c.snippets (hunk 2 above)
Comment (lpil) [inline snippets/c.snippets]: Sounds good. :)

**Comment by PavloSolntsev** [comment]
ok, this should be better. dfunX and dmfunX functions were introduced. Protected was used for class instead of private. gpl3 was removed.

**Comment by PavloSolntsev** [comment]
Did you get a chance to review my comment?

**Comment by lpil** [comment]
Sorry to have missed this!

File: snippets/c.snippets [hunk 4]
```diff
@@ -130,6 +130,84 @@ snippet fun
 	{
 		${4}
 	}
+# function definition with zero parameters
+snippet fun0
+	${1:void} ${2:function_name}()
+	{
+		${3}
```
Comment (lpil) [inline snippets/c.snippets]: I'm keen to merge this in as it has been left for a while, this improvement can come in a later PR, thanks. :)

**Pull request merged**